src/factory/index.js

```diff
@@ -0,0 +1,3 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
I like the idea of clarity around the API but am bummed by the implicit semantics around one provider/factory "containing" or encapsulating others.  For example, you might want to write a completely different service factory that uses some or all of the other providers.  That would end up violating the semantics you outlined.

More generally it would be great to minimize assumptions and have individual bits o code be stand alone with API etc.  how they get used is for the consumer.